### ISSUE TITLE
Genotype sequences?

### ISSUE THREAD

**Issue by jfass** [fritzsedlazeck/Sniffles#12]
Hi - are there any plans to include sequences in the VCF output? For example, if it's an insertion, what sequence (from the reads) has been inserted?

If this isn't planned, can you suggest other tools to extract sequence from a VCF, reference, and reads?

Thanks,
~Joe

**Comment by fritzsedlazeck** [comment]
Hi Joe,
not on short term, since this requires a de novo assembly over the reads.
I could give you the sequence from one read if that would already be helpful, but keep in mind that there are sequencing error involved. 
  
I am not aware of a tool that would do that.
  
Sorry
Fritz

**Comment by jfass** [comment]
Getting the sequence from even one read would be really helpful, in order to recruit more reads and, as you say, do de novo assembly to get a higher quality variant sequence. Then those higher quality sequences could be aligned to the reference again (with ngmlr or, since it's now much higher accuracy, other mappers), and a (presumably) more accurate variant sequence could be obtained ... it's sequence would be only "read" mapped to that location, so sniffles would give that read.

Could the read sequence have breakpoints indicated?

**Comment by fritzsedlazeck** [comment]
I thought of giving only the inserted sequence of one read or did you have a better idea?
Note that sometimes these sequences can be a bit imprecise in the breakpoint over different reads.

**Comment by jfass** [comment]
It's a good question. Maybe any whole read that covers the variant feature, or flanks both sides of it. That would cover all types of variants ... ?

The breakpoints wouldn't be accurate or necessary for the first go-round, but once someone has recruited other reads and done a local assembly, then they could be. And if sequence isn't given in the VCF, then breakpoints on a high quality read would be the next best thing.

**Comment by fritzsedlazeck** [comment]
Mhm what Sniffles already can do is the -n flag. This gives you -n X read names that support the variant.  ( https://github.com/fritzsedlazeck/Sniffles/wiki/Parameter ). You get then a list of comma separated read names in the INFO field of the vcf file with RNAMES=.

That would be a general way for every variant. Nevertheless, I think we agree that even one sequence especially for insertions can be very useful. I can try to get that in one of the next versions.

**Comment by jfass** [comment]
That'll do ... I'll experiment with using those reads to start a local assembly. But like I said that would feed back into sniffles to predict a higher accuracy variant sequence ... and I'm not aware of other software that reports the sequence of variants in a useful way yet. Would be useful!

Thanks for your help Fritz! And for the quick responses ..

**Comment by fritzsedlazeck** [comment]
I agree. Let me know how it goes. 

You can also look at the std_quant flags to see how precisely the reads agree on each other. Plus let me know if you need additional information. 
Thx
Fritz

**Comment by jfass** [comment]
What produces the 'std_quant' info ... not seeing that in the VCF output ...

**Comment by fritzsedlazeck** [comment]
Ah it is just in the latest release/version. For example:
STD_quant_start=7.691154;STD_quant_stop=21.314495;

Its in the INFO field of the VCF. Over the last weeks I focused on how we can detect if something is an SVs or an artifact.  For example we observe random insertion calls for single reads, which in high cov. region can lead to a signal. What seems to work nicely is to get the predicted breakpoint and compute the standard deviation of the observed point per read that support that SV. Next I corrected the std comp since some few outlier hide the signal. 

So overall, I would be very interested what you see when you do the de novo assembly of some questionable regions. I did not had time to do that for my own, but I think we can learn something. 

Thx
Fritz
